src/v/kafka/server/quota_manager.h

```diff
@@ -138,6 +187,7 @@ class quota_manager : public ss::peering_sharded_service<quota_manager> {
 
     client_quotas_t& _client_quotas;
     client_quota_translator _translator;
+    client_quotas_probe _probe;
```

Comment:
At some point, many of the probes were changed to have a lifetime that's limited to the `start()`, `stop()` of the service and held by reference (`std::unique_ptr<probe>`). Something about services being moved after the metrics had claimed addresses.

One advantage of that is that the probe can be forward declared, and moved to the impl.

Reply:
https://github.com/redpanda-data/redpanda/pull/19861#issuecomment-2178287587

Reply:
@pgellert - I think the other issue w/ tying probe lifetime to the service itself is that `start()` might get called twice under some circumstances (?), which could lead to name collisions (?) on successive `add_group` calls. This way we always call `~probe()` then `probe()` then `setup_metrics` with a clean slate.